### DISCUSSION TITLE
All optional object fields in the infer type

### DISCUSSION THREAD

**Discussion by awinogradov** [colinhacks/zod#426] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/982072/117222228-a7eef880-ae13-11eb-9cd2-beecc490d746.png)

`schema` is an instance of `zod`. Result must be:
```ts
type Q = {
  key: string;
  description?: string;
}
```

TS `4.2.2`

**Comment by colinhacks** [comment, marked as answer]
https://github.com/colinhacks/zod/issues/289

**Reply by awinogradov** [reply]
🔥
